### PR TITLE
Feature/1 login anbinden

### PR DESCRIPTION
**Pull request by RandomTannenbaum**

### CODE REVIEW

**Review by peggimann — CHANGES_REQUESTED** [review]

File: src/main/java/ch/puzzle/eft/config/AppServerConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                tomcat.addAdditionalTomcatConnectors(redirectConnector());
+            }
+        };
+    }
+
+    private Connector redirectConnector() {
+        Connector connector = new Connector(PROTOCOL);
+        connector.setScheme(ajpProtocol);
+        connector.setPort(ajpPort);
+        connector.setSecure(false);
```
[the remark this answers is not in the document]
Comment (RandomTannenbaum) [reply src/main/java/ch/puzzle/eft/config/AppServerConfig.java:36]: setSecure only tells the client if the connection itself is secure or not. See: https://tomcat.apache.org/tomcat-8.5-doc/api/org/apache/catalina/connector/Connector.html#setSecure(boolean)

AJP isnt a secure protocol, so I think false is okay:
https://stackoverflow.com/questions/12460422/how-do-ensure-that-apache-ajp-to-tomcat-connection-is-secure-encrypted

File: src/main/java/ch/puzzle/eft/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        authenticationProvider.setPreAuthenticatedUserDetailsService(userDetailsService);
+        return new ProviderManager(authenticationProvider);
+    }
+
+    @Bean
+    public RequestAttributeAuthenticationFilter authenticationFilter(AuthenticationManager authenticationManager) {
+        RequestAttributeAuthenticationFilter authenticationFilter = new ShibbolethRequestAuthenticationFilter(mockPrincipal);
+        authenticationFilter.setAuthenticationManager(authenticationManager);
+        authenticationFilter.setContinueFilterChainOnUnsuccessfulAuthentication(false);
+        authenticationFilter.setCheckForPrincipalChanges(false);
```
[the remark this answers is not in the document]
Comment (RandomTannenbaum) [reply src/main/java/ch/puzzle/eft/config/SecurityConfig.java:74]: When trying to access the application with this option we get "This page is not redirecting properly"

File: src/main/java/ch/puzzle/eft/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Value("${test.mock.principal:false}")
+    private boolean mockPrincipal;
+
+    @Bean
+    public AuthenticationUserDetailsService<PreAuthenticatedAuthenticationToken> userDetailsService() {
+        return token -> new AuthenticationUserDetails((AuthenticationUser) token.getPrincipal()) {
+            @Override
+            public Collection<? extends GrantedAuthority> getAuthorities() {
+                Collection<GrantedAuthority> authorities = token.getAuthorities();
```
[the remark this answers is not in the document]
Comment (RandomTannenbaum) [reply src/main/java/ch/puzzle/eft/config/SecurityConfig.java:45]: I have only ever seen an empty list of authorities in the logs and we also have no role system so afaik we dont need them currently

File: pom.xml [hunk 4]
```diff
@@ -33,40 +33,74 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-security</artifactId>
+			<version>3.3.4</version>
```
Comment (MasterEvarior) [inline pom.xml:36]: Is there a specific reason why all the spring-boot-starter versions are now explicitly mentioned? They should already be automatically configured by the ` spring-boot-starter-parent`.

**Review by MasterEvarior — APPROVED** [review]

**Pull request merged**